source/security.txt

```diff
@@ -13,13 +13,203 @@ Secure Your Data
+the relevant values for your MongoDB deployment.
+
+.. _cpp-auth-sample:
+
+.. include:: /includes/usage-examples/sample-app-intro.rst
+
+.. literalinclude:: /includes/usage-examples/connect-sample-app.cpp
+      :language: cpp
+      :copyable: true
+      :linenos:
+      :emphasize-lines: 13-15
+
+SCRAM-SHA-256
+-------------
+
+The following code shows how to create a connection URI to authenticate by using
+the ``SCRAM-SHA-256`` authentication mechanism:
+
+.. literalinclude:: /includes/authentication.cpp
+   :language: cpp
+   :copyable: true
+   :start-after: // start-scram-sha-256
+   :end-before: // end-scram-sha-256
+
+To learn more about SCRAM-SHA-256 authentication, see :ref:`cpp-scram-sha-256` in
+the Authentication guide.
+
+SCRAM-SHA-1
+-----------
+
+The following code shows how to create a connection URI to authenticate by using
+the ``SCRAM-SHA-1`` authentication mechanism:
+
+.. literalinclude:: /includes/authentication.cpp
+   :language: cpp
+   :copyable: true
+   :start-after: // start-scram-sha-1
+   :end-before: // end-scram-sha-1
+
+To learn more about SCRAM-SHA-1 authentication, see :ref:`cpp-scram-sha-1` in
+the Authentication guide.
+
+MONGODB X.509
+-------------
+
+The following code shows how to create a connection URI to authenticate by using
+the ``X.509`` authentication mechanism:
+
+.. literalinclude:: /includes/authentication.cpp
+   :language: cpp
+   :copyable: true
+   :start-after: // start-x509
+   :end-before: // end-x509
+
+To learn more about X.509 authentication, see :ref:`cpp-x509` in
+the Authentication guide.
+
+MONGODB-AWS
+-----------
+
+The following sections show how to connect to MongoDB by using the ``MONGODB-AWS``
+authentication mechanism. When you use the ``MONGODB-AWS`` mechanism, the {+driver-short+} attempts to
+retrieve your AWS credentials from the following sources, in the order listed:
+
+1. Named parameters passed to the Connection URI
+#. Environment variables
+#. AWS EKS AssumeRoleWithWebIdentity request
+#. ECS container metadata
+#. EC2 instance metadata
+ 
+Each section shows how to create the connection URI to authenticate with
+``MONGODB-AWS`` when retrieving your AWS credentials from the specified source.
+
+To learn more about authenticating with AWS, see
+:ref:`cpp-mongo-aws` in the Authentication guide.
+
+Connection URI
+~~~~~~~~~~~~~~
+
+The following code shows how to create a connection URI that includes AWS credentials to
+authenticate with ``MONGODB-AWS``:
+
+.. literalinclude:: /includes/authentication.cpp
+   :language: cpp
+   :copyable: true
+   :start-after: // start-aws-connection-uri
+   :end-before: // end-aws-connection-uri
+
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~~
+
+The following code shows how to create a connection URI to
+authenticate with ``MONGODB-AWS`` when obtaining credentials from environment variables:
+
+.. literalinclude:: /includes/authentication.cpp
```

Comment:
The "Environment Variables", "AssumeRoleWithWebIdentity Request", "ECS Metadata", and "EC2 Instance Metadata" code blocks are all identical and do not appear to demonstrate the described features as intended.

Reply:
The actual code for these is all the same, but I added some extra context regarding the variables/environment setup needed for each one. I also linked to their respective pages for more information